pysb/core.py

```diff
@@ -516,15 +518,22 @@ def is_equivalent_to(self, other):
         """Checks for equality with another ComplexPattern"""
         # Didn't implement __eq__ to avoid confusion with __ne__ operator used for Rule building
 
+        # Check both patterns are concrete
```

Comment:
I did just verify that pysb itself does only use this to compare concrete species CPs so this won't break any internals, but it *is* a breaking change to the public API -- perhaps someone else was depending on being able to compare non-concrete CPs in their model or framework. Therefore this should merely be a DeprecationWarning.

The docstring needs to change to mention the concreteness requirement, while mentioning that passing non-concrete CPs is deprecated. Suggested new docstring:

```python
"""
Test a concrete ComplexPattern for equality with another.

Use of this method on non-concrete ComplexPatterns was previously allowed, but is now deprecated.

"""